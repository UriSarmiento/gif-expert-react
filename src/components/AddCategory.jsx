// rafc para crear rapido el functional component

import { useState } from 'react';
import PropTypes from 'prop-types';

//export const AddCategory = ({setCategories}) => {
export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');
  
  const onInputChange = ({target}) => {
    setInputValue(target.value); //El target es el input como tal, entonces usamos el valor del target para actualizar el inputvalue
  };

  const onSubmit = (event) => {

    const newInputValue = inputValue.trim();

    event.preventDefault();//Para evitar que se este haciendo reset cuando hagamos submit (enter)
    if(newInputValue.length < 1) return; // Checa si el valor ingresado tiene al menos una letra. El trim quita los espacios en blanco.
    
    //setCategories( categories => [ newInputValue, ...categories ]);
    onNewCategory(newInputValue);
    setInputValue('');
  };

  // return (
  //   <form onSubmit={(event) => onSubmit(event)}>
  //     <input
  //       type = "text"
  //       placeholder = "Buscar Gifs"
  //       value= { inputValue }
  //       onChange={ (event) => onInputChange(event) }
  //     />
  //   </form>
  // );
  return (
    <form onSubmit={ onSubmit } aria-label="form">
      <input
        type = "text"
        placeholder = "Buscar Gifs"
        value= { inputValue }
        onChange={ onInputChange }
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}