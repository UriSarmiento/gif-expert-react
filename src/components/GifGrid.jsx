//import { getGifs } from "../helpers/getGifs";
import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({ category }) => {

const { images, isLoading } = useFetchGifs( category );

console.log(images);
  // Comentado porque se cambio al custom hook: useFetchGifs

  // const [images, setImages] = useState([]);

  // const getImages = async() => {
  //   const newImages = await getGifs( category );
  //   setImages(newImages);
  // }



  // // Lanzar la peticion HTTP solo cuando se cargue por primera vez
  // useEffect( () => {
  //   getImages();
  // },[] ); // Si el arreglo de dependencias se deja vacio, el hook solo se ejecuta la primera vez que se crea el componente


  return (
    <>
        <h3>{ category }</h3>
        {
          // isLoading
          // ? (<h2>Loading...</h2>)
          // : null

          isLoading && ( <h2>Loading...</h2> )
        }

        <div className="card-grid">
          {
            //images.map( ({ id, title }) => (
              images.map( ( image ) => (
             // <li key = { id }>{ title }</li>
             <GifItem key = {image.id}
                      { ...image } // asi se esparcen todas las properties
             />
            ))         
          }
       </div>
        
    </>
  );
}
