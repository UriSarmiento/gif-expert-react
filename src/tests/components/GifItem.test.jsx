import { GifItem } from "../../components/GifItem";
import { render, screen } from "@testing-library/react";

describe('Pruebas en GifItem', () => {

    const title = 'GifItem title';
    const url   = 'https://giphy.com/gifs/MDJ9IbxxvDUQM';

    test('Debe hacer match con el snapshot', () => {
        
        const {container} = render(<GifItem title = {title} url = {url} />)
        
        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        render(<GifItem title = {title} url = {url} />);

        screen.debug();

        // expect( screen.getByRole('img').src ).toBe(url);
        // expect( screen.getByRole('img').alt ).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

    test('Debe de mostrar el titulo en el componente', () => {

        render(<GifItem title = {title} url = {url} />);
        expect(screen.getByAltText(title)).toBeTruthy(); // Busca que el texto del titulo este renderizado en el componente

    });

 })