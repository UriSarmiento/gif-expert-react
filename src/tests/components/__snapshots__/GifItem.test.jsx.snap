// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="GifItem title"
      src="https://giphy.com/gifs/MDJ9IbxxvDUQM"
    />
    <p>
       
      GifItem title
       
    </p>
  </div>
</div>
`;
